syntax = "proto3";

package gpr5300.pb;

message Color4
{
    float r = 1;
    float g = 2;
    float b = 3;
    float a = 4;
}

message FrameBuffer
{

}

message SubPass
{
    Color4 clear_color = 1;
    uint32 index = 2;
}

message RenderPass
{
    repeated SubPass sub_passes = 1;
}

message Shader
{
    enum Type
    {
        VERTEX = 0;
        FRAGMENT = 1;
        COMPUTE = 2;
    }
    string path = 1;
    Type type = 2;
}

message Pipeline
{
    repeated Shader shaders = 1;
}
